### DISCUSSION TITLE
Additional swizzle for arbitrary data?

### DISCUSSION THREAD

**Discussion by mbechard** [shader-slang/slang#5586] [Ideas]
Stemming from an internal discussion we are currently having at my company. The data manipulated in shaders is often neither position/direction (xyzw) nor Color (rgba). So using those swizzles to access that data imparts incorrect information to the reader. One example is a vec2 holding something like (index, weight), and the code is accessing it via .xy.
Is there any interest/thoughts about adding an additional swizzle that is more appropriate for arbitrary data?

**Comment by csyonghe** [comment]
It appears to me that you may want to use a dedicated `struct` type to encode those things with different meaning? May I understand more detail on what's preventing you from doing:
```
struct WeightObject
{
    int index;
    float weight;
}
```

**Comment by mbechard** [comment]
Yeah, that is a workflow that would work for a lot of cases. If the data is already in a float4 though (such as coming from a texture), you can't do that.
Also the struct workflow doesn't allow for the use of swizzles.
It's not a high priority ask at all, just curious about thoughts about a generic swizzle beyond the two that are available.

**Reply by csyonghe** [reply]
You can use `reinterpret` to cast a float4 to some other type:
```
struct MyObject { int index; float v; };

void test()
{
     let object = reinterpret<MyObject>(texture.Load(uint2(0,0)).xy);
}
```

**Reply by csyonghe** [reply]
We don't have swizzle for `struct` types, however, `Tuple` types does have swizzle support.
```
Tuple<int, float, float> t;
Tuple<float,float> v = t._1_2;
```

**Comment by mbechard** [comment]
Ah great, yeah that answers it then. Seems like it's covered enough by those two cases. Thanks!
